sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareDirectoryAsyncClient.java

```diff
@@ -350,32 +353,28 @@ public Mono<Response<ShareDirectoryInfo>> createWithResponse(FileSmbProperties s
     public Mono<Response<ShareDirectoryInfo>> createWithResponse(ShareDirectoryCreateOptions options) {
         try {
             return withContext(context -> createWithResponse(options.getSmbProperties(), options.getFilePermission(),
-                options.getFilePermissionFormat(), options.getMetadata(), context));
+                options.getFilePermissionFormat(), options.getPosixProperties(), options.getMetadata(), context));
         } catch (RuntimeException ex) {
             return monoError(LOGGER, ex);
         }
     }
 
     Mono<Response<ShareDirectoryInfo>> createWithResponse(FileSmbProperties smbProperties, String filePermission,
-        FilePermissionFormat filePermissionFormat, Map<String, String> metadata, Context context) {
-        FileSmbProperties properties = smbProperties == null ? new FileSmbProperties() : smbProperties;
+        FilePermissionFormat filePermissionFormat, FilePosixProperties posixProperties, Map<String, String> metadata,
+        Context context) {
+        context = context == null ? Context.NONE : context;
```

Comment:
Can we use parenthesis here for better readability?

for ex : context = (context == null) ? Context.NONE : context;

Reply:
although I agree that it would increase readability, we don't use parenthesis for ternary operators anywhere else in the SDK, so I'm not sure it would make sense to start using them in this PR. i would absolutely be open to looking into this as a larger change across storage in the future though - I personally struggle with reading ternary operators so I think it could be helpful

Reply:
Yes, I have also noticed that we are not using parentheses anywhere throughout the SDK for ternary operators. However, it would be great if we could implement this as a larger change across storage in the future to improve readability. You can ignore it for this PR.